src/Symfony/Component/Messenger/Transport/AmqpExt/AmqpReceiver.php

```diff
@@ -110,7 +110,11 @@ public function reject(Envelope $envelope): void
      */
     public function getMessageCount(): int
     {
-        return $this->connection->countMessagesInQueues();
+        try {
+            return $this->connection->countMessagesInQueues();
+        } catch (\AMQPException $exception) {
+            throw new TransportException($exception->getMessage(), 0, $exception);
+        }
```

Comment:
Unrelated: this was just missing. We always catch transport errors and return TransportException.